provider_mock_test.go

```diff
@@ -115,9 +114,9 @@ func (m *mockProvider) GetApplication(_ context.Context, _ string) (api.Applicat
 	}, nil
 }
 
-func (m *mockProvider) DeleteApplication(_ context.Context, applicationObjectID string) (autorest.Response, error) {
+func (m *mockProvider) DeleteApplication(_ context.Context, applicationObjectID string) error {
 	delete(m.applications, applicationObjectID)
-	return autorest.Response{}, nil
+	return nil
 }
 
 func (m *mockProvider) AddApplicationPassword(_ context.Context, _ string, displayName string, endDateTime date.Time) (result api.PasswordCredentialResult, err error) {
```

Comment:
```suggestion
func (m *mockProvider) AddApplicationPassword(_ context.Context, _ string, displayName string, endDateTime date.Time) (api.PasswordCredentialResult, error) {
```

Reply:
There are other methods within `mockProvider` that use named return values, but they seem unused or unreferenced in many cases. Wondering if we can change those as well.

Reply:
I've removed all of them, but added one in: `CreateServicePrincipal` returns two strings and an error. Without the names for the two strings, it's less clear what is supposed to be returned so I added the names to help make it clearer